### PR TITLE
Upgrade orders api to Java 21

### PR DESCRIPTION
**Pull request by vshastri-ch**

### CODE REVIEW

File: pom.xml [hunk 1]
```diff
@@ -1,23 +1,24 @@
...
-        <kafka-models.version>1.0.26</kafka-models.version>
-        <ch-kafka.version>1.4.2</ch-kafka.version>
-        <java.version>1.8</java.version>
-        <structured-logging.version>1.9.12</structured-logging.version>
+        <maven.compiler.release>21</maven.compiler.release>
+        <ch-kafka.version>3.0.0</ch-kafka.version>
+        <kafka-models.version>3.0.7</kafka-models.version>
+        <java.version>21</java.version>
+        <structured-logging.version>3.0.3</structured-logging.version>
         <commons.lang.version>2.6</commons.lang.version>
```
Comment (hwatkins-ch) [inline pom.xml:22]: This is an old version as the dependency has been moved, can we move to commons-lang3 and get on the latest version of that?

https://mvnrepository.com/artifact/commons-lang/commons-lang

File: pom.xml [hunk 2]
```diff
@@ -1,23 +1,24 @@
...
         <commons.beanutils.version>1.9.4</commons.beanutils.version>
         <org.mapstruct.version>1.4.2.Final</org.mapstruct.version>
```
Comment (hwatkins-ch) [inline pom.xml:24]: Can we upgrade maptruct version to the latest, 1.5.5.FINAL?

https://mvnrepository.com/artifact/org.mapstruct/mapstruct

File: pom.xml [hunk 3]
```diff
@@ -26,28 +27,29 @@
...
+        <api-security-java.version>2.0.0</api-security-java.version>
 
-        <spring-boot-dependencies.version>2.5.5</spring-boot-dependencies.version>
-        <spring-boot-maven-plugin.version>2.5.5</spring-boot-maven-plugin.version>
+        <spring-boot-dependencies.version>3.2.3</spring-boot-dependencies.version>
+        <spring-boot-maven-plugin.version>3.2.3</spring-boot-maven-plugin.version>
 
-        <maven-compiler-plugin.version>3.8.1</maven-compiler-plugin.version>
+        <maven-compiler-plugin.version>3.12.1</maven-compiler-plugin.version>
         <maven-surefire-plugin.version>2.22.2</maven-surefire-plugin.version>
```
Comment (hwatkins-ch) [inline pom.xml:42]: Can we upgrade this maven-surefire-plugin to 3.2.5? 

https://mvnrepository.com/artifact/org.apache.maven.plugins/maven-surefire-plugin

File: pom.xml [hunk 4]
```diff
@@ -26,28 +27,29 @@
...
-        
-        <log4j-version>2.17.1</log4j-version>
+
+        <log4j-version>2.20.0</log4j-version>
 
         <!-- Docker -->
         <jib-maven-plugin.version>3.1.1</jib-maven-plugin.version>
```
Comment (hwatkins-ch) [inline pom.xml:47]: Can we upgrade jib-maven-plugin to 3.4.1?

https://mvnrepository.com/artifact/com.google.cloud.tools/jib-maven-plugin

File: pom.xml [hunk 5]
```diff
@@ -180,7 +244,7 @@
         <dependency>
             <groupId>de.flapdoodle.embed</groupId>
             <artifactId>de.flapdoodle.embed.mongo</artifactId>
-            <scope>test</scope>
+            <version>3.2.3</version>
```
Comment (hwatkins-ch) [inline pom.xml:247]: Can we move this to the properties above?

Can we also make sure its the latest version, I believe its up to 4.12.2
https://mvnrepository.com/artifact/de.flapdoodle.embed/de.flapdoodle.embed.mongo

File: pom.xml [hunk 6]
```diff
@@ -217,7 +276,7 @@
             <plugin>
                 <groupId>org.springframework.boot</groupId>
                 <artifactId>spring-boot-maven-plugin</artifactId>
-                <version>${spring-boot-maven-plugin.version}</version>
+                <version>3.2.0</version>
```
Comment (hwatkins-ch) [inline pom.xml:279]: Is there any reason we added the version here and removed it from the properties above?

This version can also be upgraded to 3.2.4
https://mvnrepository.com/artifact/org.springframework.boot/spring-boot-maven-plugin

**Review by hwatkins-ch — CHANGES_REQUESTED** [review]
Looks good, just a few dependencies that still need upgrading :)

File: pom.xml (hunk 6 above)
Comment (vshastri-ch) [reply pom.xml:279]: made it 3.2.3 and used the property

File: src/main/java/uk/gov/companieshouse/orders/api/controller/OrderController.java [hunk 7]
```diff
@@ -11,8 +11,8 @@
 import java.time.LocalDateTime;
```
Comment (foman-ch) [inline src/main/java/uk/gov/companieshouse/orders/api/controller/OrderController.java:11]: Is there an unused import here

File: src/main/java/uk/gov/companieshouse/orders/api/interceptor/RequestMapper.java [hunk 8]
```diff
@@ -1,9 +1,13 @@
...
 
 import java.util.List;
-import javax.servlet.http.HttpServletRequest;
+
+import jakarta.servlet.http.HttpServletRequest;
 import org.springframework.stereotype.Service;
+import org.springframework.web.servlet.mvc.condition.PathPatternsRequestCondition;
+import org.springframework.web.servlet.mvc.condition.RequestMethodsRequestCondition;
 import org.springframework.web.servlet.mvc.method.RequestMappingInfo;
+import org.springframework.web.util.ServletRequestPathUtils;
```
Comment (foman-ch) [inline src/main/java/uk/gov/companieshouse/orders/api/interceptor/RequestMapper.java:10]: Looks like a couple of these imports aren't used

**Review by foman-ch — COMMENTED** [review]
just some unused imports to look at

**Review by hwatkins-ch — APPROVED** [review]

**Comment by hwatkins-ch** [comment]
Just a quick one, is it worth squashing your commits? I see there are a lot of merge commits, which I don't think we want going into our main branch.

**Comment by vshastri-ch** [comment]
Will squash commits for sure. thanks!

**Review by foman-ch — APPROVED** [review]
Changes look good to me!

**Pull request merged**